src/domain/opendatafabric/src/identity/dataset_refs.rs

```diff
@@ -740,3 +741,38 @@ impl std::cmp::PartialOrd for DatasetRefAny {
         Some(self.cmp(other))
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct DatasetRefPattern {
+    pub wildcard: char,
+    pub pattern: DatasetNamePattern,
+}
```

Comment:
I think our types are not decomposed correctly.

Our CLI commands currently accept `DatasetRef`, which can be:
- `DatasetRef::ID(DatasetID("did:odf:xxx"))`
- `DatasetRef::Alias(DatasetAlias(None, DatasetName("my.dataset")))`
- `DatasetRef::Alias(DatasetAlias(Some(AccountName("alice")), DatasetName("my.dataset")))`
- `DatasetRef::Handle(DatasetHandle(id, alias))` - for internal use when a reference is already resolved to a specific dataset (to avoid expensive resolving multiple times)

Our goal is to accept **everything that was supported before** PLUS patterns.

Perhaps the simplest way to achieve all this is:
```rust
pub enum DatasetRefPattern {
  Ref(DatasetRef),
  Pattern(Option<AccountName>, DatasetNamePattern),
}
```

In our parsing function if we manage to parse a valid `DatasetRef` - we know it's just a regular reference to a single dataset. Otherwise we try to parse it as a pattern with optional account name.

Reply:
I recommend you to right away add the tests that parse all these different cases into `DatasetRefPattern`. Let's use the same examples you specified in your RFC as test inputs.